code/AssetLib/Blender/BlenderLoader.cpp

```diff
@@ -142,25 +140,23 @@ void BlenderImporter::SetupProperties(const Importer * /*pImp*/) {
 void BlenderImporter::InternReadFile(const std::string &pFile,
         aiScene *pScene, IOSystem *pIOHandler) {
     FileDatabase file;
-    std::shared_ptr<IOStream> stream(pIOHandler->Open(pFile, "rb"));
-    std::vector<char> uncompressed;
-    const std::string error = ParseMagicToken(stream, uncompressed);
-    if (!error.empty()) {
-        ThrowException(error);
+    StreamOrError streamOrError = ParseMagicToken(pFile, pIOHandler);
+    if (!streamOrError.error.empty()) {
+        ThrowException(streamOrError.error);
     }
 
     char version[4] = { 0 };
-    file.i64bit = (stream->Read(version, 1, 1), version[0] == '-');
-    file.little = (stream->Read(version, 1, 1), version[0] == 'v');
+    file.i64bit = (streamOrError.stream->Read(version, 1, 1), version[0] == '-');
```

Comment:
You should move stream into a local variable so there's less changes to existing code.

Reply:
Done.